cli/compose/template/template_test.go

```diff
@@ -81,3 +89,64 @@ func TestNonAlphanumericDefault(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, "ok /non:-alphanumeric", result)
 }
+
+func TestMandatoryVariableErrors(t *testing.T) {
+	testCases := []struct {
+		template      string
+		expectedError string
+	}{
+		{
+			template:      "not ok ${UNSET_VAR:?Mandatory Variable Unset}",
+			expectedError: "Mandatory Variable Unset",
+		},
+		{
+			template:      "not ok ${BAR:?Mandatory Variable Empty}",
+			expectedError: "Mandatory Variable Empty",
+		},
+		{
+			template:      "not ok ${UNSET_VAR:?}",
+			expectedError: "",
```

Comment:
The error here should be something like "required variable UNSET_VAR is missing a value"

Reply:
Sure. I am assuming you are referring to the error message in the test cases here, not changing the `Substitute` function, is that correct?

Reply:
No, I think the error from Substitute should be something like this:

```go
err = &InvalidTemplateError{
    Template: fmt.Sprintf("required variable %s is missing a value: %s", name, errorMessage),
}
```

For reference, this is the error in docker-compose: https://github.com/docker/compose/pull/5531/files#diff-b2372f2a81cc5911e04cc7df7d68675bR65

Reply:
Thanks for the clarification. I updated the PR (the reg. exp. also needed to be fixed to cover empty error messages)